components/raftstore/src/store/fsm/apply.rs

```diff
@@ -2744,7 +2745,7 @@ impl<EK> ApplyFsm<EK>
 where
     EK: KvEngine,
 {
-    fn from_peer<ER: KvEngine>(
+    fn from_peer<ER: RaftEngine>(
```

Comment:
Can we write like this?
```
impl<EK, ER> ApplyFsm<EK> 
where
    EK: KvEngine,
    ER: RaftEngine,
{
...
}
```

Reply:
No, because `ER` is not in the struct. We need a `PhantomData` to write code like that.